middleware/instrument.go

```diff
@@ -28,13 +29,29 @@ type RouteMatcher interface {
 	Match(*http.Request, *mux.RouteMatch) bool
 }
 
+// PerTenantCallback is a function that returns a tenant ID for a given request. When the returned tenant ID is not empty, it is used to label the duration histogram.
+type PerTenantCallback func(context.Context) string
```

Comment:
nit: `PerTenantCallback` doesn't seem accurate as a name since the function is supposed to return a tenant id. Maybe `TenantIDCallback`?

Reply:
Actually coming back to this, I'm indifferent. I think in the broader context this naming is fine.

Reply:
My thinking was more that in the metrics themselves, we call the label `tenant` and the metric `per_tenant_request_duration_seconds` and adding the "_id" wouldn't seem right in my opinion.

`per_tenant_id_request_duration_seconds`
`per_tenant_id_request_duration_seconds{tenant_id="xx"}`

So I think it is better to stick with that naming theme without ID.